pkg/filter/proxy/request.go

```diff
@@ -32,26 +33,41 @@ import (
 )
 
 type (
+	requestStatus int16
+
 	request struct {
 		server     *Server
 		std        *http.Request
 		statResult *httpstat.Result
 		createTime time.Time
 		_startTime *time.Time
 		_endTime   *time.Time
+		status     requestStatus
```

Comment:
why add this field? could it improve performance?

propose to change `_startTime` and `_endTime` from `*time.Time` to `time.Time`, this helps avoid some memory allocations, and we can use `time.Time.IsZero()` to check if they are initialized.

Reply:
It does not improve the performance. I added it, because the previous logic did not work when `_startTime` was not nil anymore after being initialized once and recycled via the sync.Pool. Thanks for hint, I will change to `time.Time` and `IsZero`.

Reply:
Well it turned out that `time.Time` is not reinitialized either when added back to pool. So the second `_startTime` does not return true for `time.Time.IsZero()`.
Here's a small Go playground example: https://go.dev/play/p/cvIuNhSe94M
One option would be manually reinitialize times, but there is no function time.Time.Zero() that would give the "_uninitialized_" value.

So I think it's better to keep this `status` field for now.

Reply:
`status` is not reinitialized either, we can reset `startTime` and `endTime` in `pool.newRequest`.

Another issue with `status`  is that it introduces some kind of redundancy to existing `startTime` & `endTime`,  this could lead to inconsistency in the code maintenance.

Reply:
time could be initialized by assigning a zero value, like:
```go
startTime = time.Time{}
```